angular/projects/global/src/lib/services/store.service.ts

```diff
@@ -0,0 +1,153 @@
+export class StoreService {
+  private currentState: any;
+
+  constructor() {
+    this.currentState = {};
+    this.state.navigationStack = [];
+  }
+
+  get state(): any {
+    return this.currentState;
+  }
+  set state(state: any) {
```

Comment:
Might be a good idea to replace `any` with some kind of state type, so consumers know to give it a `type` and `data` property.

Reply:
State can literally be anything. So you are able to set it to an array, form, string, custom object, etc.